### PR TITLE
First pass at Coupon support

### LINKED ISSUE #978: Coupon Codes in Signup Links
### Epic

_No response_

### Description

**As a:** user

**I want to:** be able to embed the coupon code in the signup URL and have that auto-populate/apply the coupon code at the stripe checkout

**So that:** I can make better use of coupon code

### Which customers would this be availble to

_No response_

### Acceptance Criteria

- [ ] Signup URL can contain a query param eg `/signup?coupon=SOMECODE`
- [ ] This is persisted (cookie?) through the users signup journey and when they are redirected to the Stripe checkout sesison the coupon is passed in the call to stripe.
- [ ] Coupon is automatically applied to the subscription
- ...

### PR DESCRIPTION
**Pull request by hardillb**
http://forge.example.com/account/create?code=[CouponCode]

Sets the `ff_coupon` cookie with a week long life

closes #978

### CODE REVIEW

**Comment by hardillb** [comment]
OK, works, but might want a little UI polishing, I've picked what I think are the right classes, but the spacing might be a little off

**Review by sammachin — APPROVED** [review]

File: frontend/src/pages/team/Billing.vue [hunk 1]
```diff
@@ -97,13 +105,18 @@ export default {
...
-            }]
+            }],
+            coupon: false,
+            errors: {}
         }
     },
     watch: { },
     async mounted () {
+        // TODO remove
+        console.log(await window.cookieStore.get('ff_coupon'))
```
Comment (knolleary) [inline frontend/src/pages/team/Billing.vue:116]:
```suggestion
```

File: frontend/src/pages/team/Billing.vue [hunk 2]
```diff
@@ -116,8 +129,9 @@ export default {
                 })
                 this.loading = false
             } catch (err) {
-                // check for 404 and redirect if 404 returned
-                if (err.response.status === 404) {
+                // check for 402 and redirect if 402 returned
+                if (err.response.status === 402) {
+                    this.coupon = (await window.cookieStore.get('ff_coupon'))?.value.split('.')[0]
```
Comment (knolleary) [inline frontend/src/pages/team/Billing.vue:134]: `window.cookieStore` is not supported widely enough for us to use - https://developer.mozilla.org/en-US/docs/Web/API/CookieStore#browser_compatibility

**Review by knolleary — CHANGES_REQUESTED** [review]
A couple initial comments without looking at the visuals of it.

File: frontend/src/pages/team/Billing.vue (hunk 2 above)
Comment (hardillb) [reply frontend/src/pages/team/Billing.vue]: Pushed update without `window.cookieStore`

File: frontend/src/pages/team/Billing.vue [hunk 3]
```diff
@@ -133,12 +145,31 @@ export default {
...
                 }
             }
             window.open(billingUrl, '_self')
+        },
+        getCookie (name) {
+            if (document.cookie) {
+                const cookies = document.cookie.split(';')
+                for (let i = 0; i < cookies.length; i++) {
+                    const cookie = cookies[i]
+                    if (cookie.split('=')[0] === name) {
```
Comment (knolleary) [inline frontend/src/pages/team/Billing.vue:167]: Given multiple cookies, the `;` is space-padded between entries so need to trim.

```suggestion
                    if (cookie.split('=')[0].trim() === name) {
```

**Review by knolleary — APPROVED** [review]

**Pull request merged**